src/__tests__/helpers.js

```diff
@@ -97,4 +97,16 @@ describe('run with real timers', () => {
     })
     expect(global.setTimeout).toBe(fakedSetTimeout)
   })
+
+  test('safe check for setImmediate and clearImmediate', () => {
+    const setImeediate = global.setImmediate
+    const clearImmediate = global.clearImmediate
+    delete global.setImmediate
+    delete global.clearImmediate
+
+    expect(() => runWithRealTimers(() => {})).not.toThrow()
+
+    global.setImmediate = setImeediate
+    global.clearImmediate = clearImmediate
+  })
```

Comment:
It would add a bit more ceremony but make sure that we restore deleted globals in case this test fails e.g by wrapping in `describe` and adding `beforeEach` and `afterEach`.

Otherwise a failure of this test might break other tests even if they're passing. This could result in debugging the wrong test.

Reply:
I really thought that it had something wrong here, thanks for the suggestion.